src/services/textChanges.ts

```diff
@@ -446,38 +525,51 @@ namespace ts.textChanges {
         }
 
         private computeNewText(change: Change, sourceFile: SourceFile): string {
-            if (!change.node) {
+            if (change.kind === ChangeKind.Remove) {
                 // deletion case
                 return "";
             }
+
             const options = change.options || {};
-            const nonFormattedText = getNonformattedText(change.node, sourceFile, this.newLine);
+            let text: string;
+            const pos = change.range.pos;
+            const posStartsLine = getLineStartPositionForPosition(pos, sourceFile) === pos;
+            if (change.kind === ChangeKind.ReplaceWithMultipleNodes) {
+                const parts = change.nodes.map(n => this.getFormattedTextOfNode(n, sourceFile, pos, options));
+                text = parts.join(change.options.nodeSeparator);
+            }
+            else {

```

Comment:
>else { [](start = 12, length = 6)

Would it be worthwhile to assert the kind so that we get some kind of warning if a new one is added?